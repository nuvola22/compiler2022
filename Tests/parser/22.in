var a, b: Integer;

function max(num1, num2: integer): integer;
var result: integer;
begin
   if num1 = num2 then result := num1
   else result := num2;
   max := result;
end;

begin
    a := 10;
    a += 1;
    a -= 2;
    a *= 3;
    a /= 4;
    b := max(0, 10);
end.