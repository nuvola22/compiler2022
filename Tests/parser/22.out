      program
         variable
            a
            Integer
         variable
            b
            Integer
         max
            parameters:
               num1
               integer
               num2
               integer
            declarations:
               variable
                  result
                  integer
            type:
               integer
            begin
               if
                  exp:
                     =
                        num1
                        num2
                  body:
                     :=
                        result
                        num1
                  else:
                     :=
                        result
                        num2
               :=
                  max
                  result
         begin
            :=
               a
               10
            +=
               a
               1
            -=
               a
               2
            *=
               a
               3
            /=
               a
               4
            :=
               b
               call
                  max
                  0
                  10
